README.md

```diff
@@ -1,13 +1,237 @@
+- Autogenerate types based on your schema. These types can be imported in your
+  sources to write safer code (Typescript only for now).
+
+These benefits help you to catch inconsistencies, exposed secrets, and invalid
+configs as early as possible in the development process. To do so, you will use
+widely adopted tools you are probably already familiar with: JSON (or YAML),
+your cloud provider, and yet another npm package. Try it out!
+
+## Example
+
+Here is how an encrypted config file containing secrets looks like:
+
+```yml
+logger:
+  level: DEBUG
+auth:
+  url: my-auth-url
+  myApiKeySecret: ENC[AES256_GCM,data:aeQ+hlVIah7WyJoVR/Jbkb6GLH7ihsV0D81+U++pkiWD0zeoRL/Oe9Q3Tz6j/TNvKKVDnohIMyw3UVjELOuSY+A==,iv:nVRZWogV4B7o=,tag:KrE2jssfP4uCvqq+pc/JyQ==,type:str]
+sops:
+  gcp_kms:
+  -   resource_id: projects/my-project/locations/europe-west3/keyRings/my-project-key-ring/cryptoKeys/my-strong-config-key
+      created_at: '2019-11-07T10:11:12Z'
+      enc: AiAAmdAgjOdw1XdV2MsDpvmZ4Deo867hmcX2B3DNhe2BCF2axuZ18hJJFK9oBlE1BrD70djwqi+L8T+NRNVnGUP+1//w8cJATAfJ8W/cQZFcdFTqjezC+VYv9xYI8i1bRna4xfFo/INIJtFDR38ZH1nrQg==
+  lastmodified: '2019-11-07T10:11:12Z'
+  mac: ENC[AES256_GCM,data:ABcd1EF2gh3IJKl4MNOpQr5stuvWXYz6sBCDEfGhIjK=,iv:A1AaAAAaa111a1Aa111AA/aaaAaaAAaa+aAaAaAAAaA=,tag:AAaaA1a1aaaAa/aa11AaaA==,type:str]
+  encrypted_suffix: Secret
+  version: 3.4.0
+```
+
+The field `sops` contains a set of metadata provided context information and
+important details to decrypt this config at runtime. Never delete this field
+or make changes to the encrypted file without using `@strong-config` - it will
+leave you with a corrupted config and a world full of terror.
 
 ## How-to
 
-Set the environment variable `RUNTIME_ENVIRONMENT`
+Please backup your existing configs and secrets_
+
+1. Install `@strong-config/node` in your project
+
+    ```sh
+    npm install @strong-config/node
+    yarn add @strong-config/node
+    ```
+
+1. Define runtime environment and create first config file
+
+    ```sh
+    export NODE_ENV=development # or similar, depending on your shell
+
+    # inside your project
+    mkdir config
+    echo "someField: '💪'" > config/development.yaml
+    ```
+
+    `strong-config` uses an environment variable to determine which config file
+    to load. By default, this is `NODE_ENV`. Setting `NODE_ENV=development` will
+    eventually load `development.yaml` from the config directory.
+
+1. Import and use `strong-config` in code
+
+    ```js
+    // index.js, or anywhere you need config values
+
+    // ES6-style import. Commonjs is supported too
+    import StrongConfig from '@strong-config/node'
+
+    // uses default parameters
+    const config = (new StrongConfig()).load()
+
+    console.log(config)
+    // --> { someField: '💪' }
+    ```
+
+    The first invocation of `load()` dynamically generates types for Typescript and
+    saves them to `./strong-config.d.ts`. If you do not use Typescript or despise
```

Comment:
I'd remove the "or despise types because of other reasons"

Reply:
done